### ISSUE TITLE
Move 'Add new/existing field' functionality to a new page via an action link

### ISSUE THREAD

**Issue by BWPanda** [backdrop/backdrop-issues#4898]
**Description of the need**

The Manage Fields page (`/admin/structure/types/manage/[TYPE]/fields`) displays a list of fields for a content type, and then two rows at the bottom for adding new/existing fields.

![image](https://user-images.githubusercontent.com/2385329/105437669-94c97a00-5cb5-11eb-889e-48f136c7bdf2.png)

This UI pattern isn't seen in many other places in core. One exception is the Roles page (`/admin/config/people/roles`), but we already have [an issue](https://github.com/backdrop/backdrop-issues/issues/2742) for changing that too. Instead, adding a new item to a list of items is done via an action link above the table. For examples, see any of the following pages:
- Text editors and formats
- Image styles
- Date and time formats
- URL aliases
- URL redirects

![image](https://user-images.githubusercontent.com/2385329/105438089-72842c00-5cb6-11eb-9599-2a587a4b002b.png)

**Proposed solution**

I recommend making the same changes here as for other places in core - removing the two 'add' rows and instead adding a new 'Add field' action link. That link would take the user to a new page where they can add a new or existing field. This will not only provide more consistency throughout core's UI, but will also help to make the display of labels & machine names more consistent, since the current UI doesn't allow for https://github.com/backdrop/backdrop-issues/issues/3402 to change this page.

For *this* issue, I propose leaving the field UI more or less as-is, for the purposes of getting this done quickly and easily. We can leave the total re-design of the field UI to other issues like https://github.com/backdrop/backdrop-issues/issues/779 (for which this issue will make a good first step).

**Additional information**

Drupal also made this change in 8.0.x: https://www.drupal.org/node/2393181

**Draft of feature description for Press Release (1 paragraph at most)**

The interface for adding fields to content types is now more consistent with the rest of Backdrop. A new 'Add field' action link takes users to a separate page where they can add new or existing fields to their content type. This helps make the Manage Fields page simpler and more focused on displaying and organising existing fields.

**Comment by klonos** [comment]
I'm supportive of this 👍 

Some screenshots of how this works in D8/9:

![image](https://user-images.githubusercontent.com/2423362/105473611-f8e44200-5ca5-11eb-90e2-6f625d2d2f98.png)

...clicking the "+ Add field" button, takes you to another page, where you can either select to add a new field type:

![image](https://user-images.githubusercontent.com/2423362/105473808-32b54880-5ca6-11eb-8f26-48886c00a6c6.png)

...or select one of the existing fields:

![image](https://user-images.githubusercontent.com/2423362/105473916-57a9bb80-5ca6-11eb-9976-7cb14d6d0877.png)

The rest of the workflow from there on is pretty much the same as current.

**Comment by indigoxela** [comment]
This might work, but... People might be missing the "Add existing field" section. I'm not sure if they realize that the "Add field" link now covers both. At first sight - I wouldn't.

**Comment by klonos** [comment]
We could make the label of the link be "+ Add new/existing field", but I'd much rather not (too wordy) 🤔

**Comment by klonos** [comment]
...or we could have 2 links: "+ Add new field" / "+ Add existing field"

**Comment by indigoxela** [comment]
> "+ Add new/existing field"

This doesn't seem that wordy to me. (But I usually can't contribute much to UI discussions, anyway. :wink:)

> 2 links: "+ Add new field" / "+ Add existing field"

That might be odd, if on that second page the decision has to be made again (two dropdowns).

**Comment by BWPanda** [comment]
> That might be odd, if on that second page the decision has to be made again (two dropdowns).

I wonder if @klonos meant for the two links to point to the same page (I doubt it) or if he was suggesting two different forms/pages - one for adding a new field and one for adding an existing field...

**Comment by olafgrabienski** [comment]
In my opinion, it would be good to see the two action links (Add new field / Add existing field) right away (and not on the second page). Seeing the other fields is a good foundation to decide if I want a new field or rather an existing one.

**Comment by klonos** [comment]
I'm taking a crack at this. Trying to get 2 "+ Add" links, one for new, and another for existing fields. This means also 2 separate forms for each of these tasks (with separate validation/submit functions for each), to keep things tidy. Once I have a working PR in place, people can start playing with the sandbox, and we can reassess whether 2 separate links/forms makes for better UX than a single link/form to perform both of these tasks.

**Comment by philsward** [comment]
+1 for "type to find" select list search box

**Comment by klonos** [comment]
I never got around to this 😓  ...up for grabs if anyone is interested.

**Comment by herbdool** [comment]
There are some related discussions here: https://github.com/backdrop/backdrop-issues/issues/647#issuecomment-675673436 and https://github.com/backdrop/backdrop-issues/issues/779#issuecomment-163844301.

E.g. mockups from those comments:

![Screen Shot 2020-08-18 at 10 24 11 pm](https://user-images.githubusercontent.com/2423362/90556526-c5a30180-e1a1-11ea-8095-0ef5eb5b0241.png)

and (step 2 of https://github.com/backdrop/backdrop-issues/issues/779#issuecomment-163844301):

![field ui - step 2](https://cloud.githubusercontent.com/assets/5607236/11736411/1a7fffca-9f9c-11e5-91ba-5ed1e4053a5a.png)
